### PR TITLE
Add shadowColor to AppBar and AppBarTheme

### LINKED ISSUE #27461: Theme should be able to set the shadow color
The Material widget has the shadowColor property which defaults to 100% black. Many UI widgets are composed with this default material (like AppBar, Card, different buttons). However, on some light designs such dark shadow is too much. Also, we could achieve an app theme with tinted shadows.

It would be awesome if we could have a 'shadowColor' property inside the ThemeData. And the Material widget should default its shadowColor to its value.

### PR DESCRIPTION
**Pull request by anisalibegic**
## Description

This PR exposes the `shadowColor` parameter of `AppBar` and `AppBarTheme` so that users can specify the color of shadow generated by elevation just like they can when using `Material` widget.

**Before this PR:**
```
Material(
  child: AppBar(
    title: Text('Title'),
    elevation: 0.0,
  ),
  elevation: 8.0,
  shadowColor: Colors.red,
)
```

**After this PR:**
```
AppBar(
  title: Text('Title'),
  elevation: 8.0,
  shadowColor: Colors.red,
)
```

## Related Issues

This PR fixes https://github.com/flutter/flutter/issues/27461 partially.

## Tests

I added the following tests:
- [AppBar.shadowColor takes priority over AppBarTheme.shadowColor](https://github.com/anisalibegic/flutter/commit/5c858c848af14f3ec1968ef9625614478a21c62b#diff-f4288da437a1629d01437d50cdc31468R236)

## Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by anisalibegic** [comment]
@googlebot I fixed it.

**Comment by anisalibegic** [comment]
> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors. If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA ([login here](https://cla.developers.google.com/) to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.
> In order to pass this check, please resolve this problem and then comment `@googlebot I fixed it.`. If the bot doesn't comment, it means it doesn't think anything has changed.
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fflutter%2Fflutter%2Fpull%2F58708) for more info**.

I fixed it.

**Comment by anisalibegic** [comment]
@googlebot I fixed it.

**Comment by anisalibegic** [comment]
Mistakenly closed. Reopened...

**Comment by anisalibegic** [comment]
@HansMuller Can you review this PR? I hope I tagged the right person, according to the [tree hygiene](https://github.com/flutter/flutter/wiki/Tree-hygiene#who-to-have-review-your-code) document.

**Comment by Piinks** [comment]
Hi @anisalibegic welcome! And thank you for the contribution! 🎉 
We may want to incorporate this as part of the ThemeData as the conversation in the issue suggests. @HansMuller has been doing a lot of work in Theming, WDYT Hans?

File: packages/flutter/test/material/app_bar_theme_test.dart [hunk 1]
```diff
@@ -219,12 +226,28 @@ void main() {
     // the value of NavigationToolBar.centerMiddle should be true.
     expect(navToolBar.centerMiddle, true);
   });
+
+  testWidgets('AppBar.shadowColor takes priority over AppBarTheme.shadowColor', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(appBarTheme: const AppBarTheme(shadowColor: Colors.red)),
+      home: Scaffold(
```
Comment (HansMuller) [inline packages/flutter/test/material/app_bar_theme_test.dart:233]: The formatting is slightly off here. Should be:

```dart
home: Scaffold(
  appBar: AppBar(
    title: const Text('Title'),
    shadowColor: Colors.yellow,
  ),
),
```

**Review by HansMuller — APPROVED** [review]
LGTM

This is a nice change, well done!

**Review by anisalibegic — COMMENTED** [review]
I fixed formatting of my PR and of some previous code which had similar formatting issue like mine.

**Comment by HansMuller** [comment]
Thanks for your contribution! It will land when the build is green again.

**Comment by anisalibegic** [comment]
> Thanks for your contribution! It will land when the build is green again.

Thank you very much!

**Pull request merged**